### PR TITLE
Initial commits

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch

### CODE REVIEW

**Comment by raymondfeng** [comment]
Connect to strongloop-internal/scrum-loopback#423

**Comment by raymondfeng** [comment]
@ritch Ping.

**Comment by rmg** [comment]
@slnode test please

File: server/middleware/acl.js [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+  } else {
+    return p2.length > p1.length ? 1 : -1;
+  }
+}
+
+/**
+ * Check the http request against route based ACLs
+ *
+ * ```js
+ * app.use(loopback.acl(
```
Comment (ritch) [inline server/middleware/acl.js:356]: Can we remove this comment?

File: server/middleware/acl.js [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+}
+
+/**
+ * Calculate the ACL score based on its specifics of principalType and
+ * permission
+ * @param {Object} acl ACL rule
+ * @returns {number}
+ * @private
+ */
+function getACLScore(acl) {
```
Comment (ritch) [inline server/middleware/acl.js:240]: Does this duplicate the algorithm in loopback core? If so we should have an issue tracking removing that duplication...

**Comment by ritch** [comment]
Left a few comments, but otherwise LGTM.

File: server/middleware/acl.js (hunk 1 above)
Comment (raymondfeng) [inline server/middleware/acl.js:356]: Can you explain why?

**Pull request merged**